controller/status.go

```diff
@@ -17,33 +18,77 @@ var (
 	StartTime = time.Now().UTC().Format("2006-01-02T15:04:05Z")
 )
 
+type statusConfiguration interface {
+	IsPostgresDeveloperModeEnabled() bool
+	DefaultConfigurationError() error
+}
+
+// DBChecker is to be used to check if the DB is reachable
+type DBChecker interface {
+	Ping() error
+}
+
 // StatusController implements the status resource.
 type StatusController struct {
 	*goa.Controller
-	db *gorm.DB
+	dbChecker DBChecker
+	config    statusConfiguration
 }
 
 // NewStatusController creates a status controller.
-func NewStatusController(service *goa.Service, db *gorm.DB) *StatusController {
+func NewStatusController(service *goa.Service, dbChecker DBChecker, config statusConfiguration) *StatusController {
 	return &StatusController{
 		Controller: service.NewController("StatusController"),
-		db:         db,
+		dbChecker:  dbChecker,
+		config:     config,
 	}
 }
 
 // Show runs the show action.
 func (c *StatusController) Show(ctx *app.ShowStatusContext) error {
-	res := &app.Status{}
-	res.Commit = Commit
-	res.BuildTime = BuildTime
-	res.StartTime = StartTime
+	res := &app.Status{
+		Commit:    Commit,
+		BuildTime: BuildTime,
+		StartTime: StartTime,
```

Comment:
( Now that we are refactoring ) 
Since we are doing the db test + configuration tests, should we report that in the status response too ?

In fact, instead of just returning the error in the response ( when there's an error ), could we stick to this format and specify the error in the json with the appropriate http response code.
example,

```
{
  "commit":"xyz123",
  "build_time":"xyz123",
  "start_time":"xyz123",
   "database_status":"Error connecting to host",
   "configuration_status":"Error: default configuration is being read."
}
```

This will help us confirm which is the build/commit which is breaking.
What do you think?

Reply:
We do return the whole json with the commit hash, build time, start time and error message in case of errors. But we return it as 503 instead of 200 if there is no error.

But we could refactor our error key in that JSON and use db_status + config_status instead. Let’s do it.

Reply:
Done in https://github.com/fabric8-services/fabric8-auth/pull/206/commits/e9a711bb60558fe6b9383d268ce559da7c23f945

Reply:
I also updated the description of the PR above. Added examples of the new status output.